sycl/test-e2e/README.md

```diff
@@ -379,3 +379,104 @@ Note: please avoid using `REQUIRES: TEMPORARY_DISABLED` for this purpose, it's
 a non-standard mechanism. Use `UNSUPPORTED: true` instead, we track
 `UNSUPPORTED` tests using the mechanism described above. Otherwise the test
 risks remaining untraceable.
+
+## Compiling and executing tests on separate systems
+
+The execution of e2e tests can be separated into compilation and execution 
+stages via the `test-mode` lit parameter. This allows us to reduce testing time 
+by compiling tests on more powerful systems and reusing the binaries on other
+machines. By default the `test-mode` parameter is set to `full`, indicating 
+that both stages will run. This parameter can be set to `build-only`, or 
+`run-only`, to only run the compilation stage, or the execution stage 
+respectively.
+
+The modes work as follow:
+### `--param test-mode=full`
+This is the default mode tests run in. Tests are marked as unsupported if no
+device on the machine can fulfill the `REQUIRES`/`UNSUPPORTED` statements. In
+this mode all `RUN:` lines are executed normally, and two extra features are
+added: the `build-and-run-mode` and `run-mode`.
+
+To make a test only run in `full` mode add a `REQUIRES: build-and-run-mode` line.
+
+### `--param test-mode=build-only`
+This mode can be used to compile all test binaries. To do this all `UNSUPPORTED` 
+and `REQUIRES` statements are ignored unless they contain `UNSUPPORTED: true` or 
+`REQUIRES: build-and-run-mode` (NOTE: currently a WIP to make this mode properly 
+react to features). All `RUN:` lines within a test are ran in this mode unless 
```

Comment:
> NOTE: currently a WIP to make this mode properly react to features

Can we drop WIP stuff from this PR and add it to the doc once it is implemented? I personally prefer to have docs in sync with the current implementation.

Reply:
Alternatively, we can hold off this PR till your WIP PR gets merged, if you are expecting it to be merged soon.

Reply:
I removed the part you highlighted. However I kept the portion on "Falling back to `full` testing mode on `run-only`", since I think that can/should be added before proceeding with the CI changes as well as this.